lib/cc/config/yaml_adapter.rb

```diff
@@ -0,0 +1,72 @@
+module CC
+  class Config
+    class YAMLAdapter
+      DEFAULT_PATH = ".codeclimate.yml".freeze
+
+      attr_reader :config
+
+      def self.load(path = DEFAULT_PATH)
+        new(::YAML.safe_load(File.read(path)))
+      end
+
+      def initialize(yaml = {})
+        @config = yaml || {}
+
+        upconvert_legacy_yaml!
+      end
+
+      private
+
+      def coerce_engine(data)
+        if [true, false].include?(data)
+          { "enabled" => data }
+        elsif data.is_a?(Hash)
+          data
+        else
+          {}
+        end
+      end
+
+      # Many of our plugins still expect:
+      #
+      #   { config: PATH }
+      #
+      # But we document, and hope to eventually move to:
+      #
+      #   { config: { file: PATH } }
+      #
+      # We need to munge from the latter to the former when/if we encounter it
+      def convert_to_legacy_file_config(config)
+        if config.is_a?(Hash) && config.keys.size == 1 && config.key?("file")
```

Comment:
Might be able to use `config.keys.one?` here.

Reply:
Changed, good catch.